APMrover2/mode_steering.cpp

```diff
@@ -12,7 +12,7 @@ void ModeSteering::update()
     // acceleration at full steering lock for this speed. That is V^2/R where R is the radius of turn.
```

Comment:
@peterbarker,. perhaps a calc_lateral_acceleration_max(float speed) would make sense.  I'd say that's out-of-scope for this PR though because that line was basically already there.  I just removed the dependency on pulling it out of the steering controller based on the P gain of the rate-ish controller that was there before.  I'm somewhat dubious that it was a valid way to get the turn-radius but perhaps the designer of old controller would know better.